### PR TITLE
#150001924 Dev should be able to bookmark a new contribution

### PR DESCRIPTION
**Pull request by andela-ieyo**
**What does this PR do?**
   - create the post bookmark view
   - Add /bookmarks to the project endpoints
   

#### What are the relevant pivotal tracker stories?
     #150001924

### CODE REVIEW

**Comment by njirap** [comment]
What is the meaning of the task `Add contribution_type to new contributions` under `What does PR do` in the description above?

File: api/models.py [hunk 1]
```diff
@@ -146,9 +154,47 @@ class Path(db.Model, ModelOpsMixin):
...
+    """
+    The Bookmark Model
+    """
+
+    __tablename__ = "Bookmark"
+
+    id = db.Column(db.String(), primary_key=True)
+    contribution_id = db.Column(db.String(), db.ForeignKey('Contribution.id'))
+    path_id = db.Column(db.String(), db.ForeignKey('Path.id'))
+    created_at = db.Column(db.DateTime, default=datetime.utcnow())
```
Comment (Andela-ddiei) [inline api/models.py]: Remove the parentheses from datetime.utcnow

File: api/views/bookmarks.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        if not _user:
+            return jsonify(dict(status=404,
+                                    message='User not found'))
+
+        if not request.json:
+            return {"message": "Request must be a valid JSON"}, 400
+
+        payload = request.get_json()
+
+        bookmark_id = '-K' + str(uuid4().hex).upper()[0:6]
```
Comment (andela-uibrahim) [inline api/views/bookmarks.py]: you may want to update this ID with the new engineering ID format

File: api/views/bookmarks.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        resource_type = ''
+
+        is_bookmarked = False
+
+        if 'contributionId' in payload.keys():
+            _contribution = Bookmark.query.filter_by(contribution_id=payload['contributionId']).first()
```
Comment (andela-uibrahim) [inline api/views/bookmarks.py]: hey i am kinda confused here, how is this user's bookmarks?

**Pull request closed without merging**